src/main/java/org/aksw/simba/squirrel/deduplication/hashing/impl/IntervalBasedMinHashFunction.java

```diff
@@ -0,0 +1,54 @@
+package org.aksw.simba.squirrel.deduplication.hashing.impl;
+
+import org.aksw.simba.squirrel.deduplication.hashing.HashValue;
+import org.aksw.simba.squirrel.deduplication.hashing.RDFHashFunction;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.graph.Triple;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Use Min hash algorithm to compute hash values for triples.
+ */
+public class IntervalBasedMinHashFunction implements RDFHashFunction {
+
+    /**
+     * The number n for the 2^n Intervals for the Hashing.
+     */
+    private int powerNumberOfIntervals;
+
+    /**
+     * Constructor.
+     *
+     * @param powerNumberOfIntervals The powered number of Intervalls for the hashing.
+     */
+    public IntervalBasedMinHashFunction(int powerNumberOfIntervals) {
+        this.powerNumberOfIntervals = powerNumberOfIntervals;
+    }
+
+    @Override
+    public HashValue hash(List<Triple> triples) {
+        List<Integer> listHashValues = new ArrayList<>(powerNumberOfIntervals);
+        for (int i = 0; i < powerNumberOfIntervals; i++) {
+            listHashValues.add(i, null);
+        }
+        int intervalSize = 2 * Integer.MAX_VALUE / (int) Math.pow(2, powerNumberOfIntervals);
```

Comment:
I am not sure whether this computation works as it should since `2 * Integer.MAX_VALUE` is already not manageable within an `int` value. Apart from the fact that `Integer.MAX_VALUE` is not the number that you would like to have here. Please keep in mind how negative numbers look like and that the left most bit is reserved for these numbers.
Btw. this would be much faster and saver when you would use bit shifting ;)